provider/datagen/src/baked_exporter.rs

```diff
@@ -509,108 +533,140 @@ impl DataExporter for BakedExporter {
             }
         };
 
-        self.write_to_file(
-            PathBuf::from("macros"),
-            quote! {
-                /// Implement [`DataProvider<M>`](icu_provider::DataProvider) on the given struct using the data
-                /// hardcoded in this file. This allows the struct to be used with
-                /// `icu`'s `_unstable` constructors.
-                ///
-                /// ```compile_fail
-                /// struct MyDataProvider;
-                /// include!("/path/to/generated/macros.rs");
-                /// impl_data_provider(MyDataProvider);
-                /// ```
-                #[doc(hidden)]
-                #[macro_export]
-                macro_rules! __impl_data_provider {
-                    ($provider:path) => {
-                        #(
-                            #features
-                            #macro_idents ! ($provider);
-                        )*
-                    }
-                }
-                #[doc(inline)]
-                pub use __impl_data_provider as impl_data_provider;
+        let any_body = self.tokens_to_string(any_body, true, 4)?;
+
+        // This function uses string interpolation instead of quote!() since
+        // it rustfmt cannot format inside macro bodies (on stable, there is a nightly
```

Comment:
Does it actually work on nightly? Maybe we should just use that

Reply:
Yes, it seems to work fine.

@sffc thoughts on having databake call `rustfmt +nightly`? This will break on systems not using `rustup`

(We can potentially allow for an env var override)

Reply:
Actually, huh, that option works on stable too.

Reply:
I guess "unstable" in rustfmt's context does not mean "unavailable on stable"

Reply:
Though [the docs claim](https://rust-lang.github.io/rustfmt/?version=v1.5.2&search=):

```
Each configuration option is either stable or unstable. Stable options can always be used, while unstable options are only available on a nightly toolchain and must be opted into. To enable unstable options, set unstable_features = true in rustfmt.toml or pass --unstable-features to rustfmt.
```

The line about it only being available on nightly is gone on the v2.0-rc docs, but it still says you need to opt in to unstable options. Unsure what's going on here.

Reply:
Clients probably won't use `pretty` on a regular basis because it's slow, we mainly want it for development.

Reply:
Seems to work with older compilers too. Hmm

Reply:
https://github.com/rust-lang/rustfmt/pull/4228

Reply:
https://github.com/unicode-org/icu4x/pull/3491

Review and merge whichever one you prefer.